File: app/src/main/java/com/dian/demo/repository/local/DataBaseManager.java
package com.dian.demo.repository.local;

import android.content.Context;


public class DataBaseManager {


    private static volatile DataBaseManager mInstance;

    private Context mContext;


    private DataBaseManager(Context mContext) {
        this.mContext = mContext;
    }

    public static DataBaseManager getInstance(Context mContext) {
        if (mInstance == null) {
            synchronized (DataBaseManager.class) {
                if (mInstance == null) {
                    mInstance = new DataBaseManager(mContext);
                }
            }
        }
        return mInstance;
    }

    /**
    *  样例
    *  public void deleteAllDatabaseEntities() {
    *         workService.execute(() -> {
    *             DatabaseFactory.getInstance(mContext).getDatabase().deleteAllDatabaseEntities();
    *         });
    *     }
    */


}

File: app/src/main/java/com/dian/demo/di/repository/local/DataBaseManager.kt
package com.dian.demo.di.repository.local

import android.content.Context
import com.dian.demo.ProjectApplication
import com.dian.demo.di.model.ArticleEntity
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

object DataBaseManager{

    suspend fun insertArticleEntities(articleEntities: List<ArticleEntity>) {
       withContext(Dispatchers.IO){
            DatabaseFactory.getInstance(ProjectApplication.getAppContext()).getArticle().insertArticleEntities(articleEntities)
        }
    }

    suspend fun deleteArticleEntities() {
        withContext(Dispatchers.IO){
            DatabaseFactory.getInstance(ProjectApplication.getAppContext()).getArticle().deleteAllArticleEntities()
        }
    }

}